dvc/stage.py

```diff
@@ -339,6 +339,7 @@ def _stage_fname(cls, fname, outs, add):
     def _expand_to_path_on_add_local(add, fname, out, path_handler):
         if (
             add
+            and out.scheme == "local"
             and out.is_local
```

Comment:
@efiop , the `out.is_local` shouldn't check this? if it is deprecated, let's remove this line

Reply:
@mroutis `is_local` is about local output being inside of a dvc repo. But yeah, makes sense to make all non-local outputs be outside of repo. I'll rename it to be less confusing as well. Thanks for pointing that out! :slightly_smiling_face: